src/react/components/pages/homepage/homePage.tsx

```diff
@@ -116,8 +121,20 @@ export default class HomePage extends React.Component<IHomepageProps> {
     }
 
     private onProjectFileUpload = (e, projectJson) => {
```

Comment:
I would like to see a test for this scenario to verify your showError is called when the user attempt to upload a invalid file.